pkg/xds/bootstrap/generator.go

```diff
@@ -95,35 +97,43 @@ func (b *bootstrapGenerator) Generate(ctx context.Context, request types.Bootstr
 	if params.ProxyType == "" {
 		params.ProxyType = string(mesh_proto.DataplaneProxyType)
 	}
-	if request.AdminPort != 0 {
-		params.AdminPort = request.AdminPort
-	}
 
 	switch mesh_proto.ProxyType(params.ProxyType) {
 	case mesh_proto.IngressProxyType:
 		zoneIngress, err := b.zoneIngressFor(ctx, request, proxyId)
 		if err != nil {
 			return nil, err
 		}
-		// The admin port in kuma-dp is always bound to 127.0.0.1
-		if zoneIngress.UsesInboundInterface(core_mesh.IPv4Loopback, params.AdminPort) {
-			return nil, errors.Errorf("Resource precondition failed: Port %d requested as both admin and inbound port.", params.AdminPort)
-		}
+
 		params.Service = "ingress"
+		if adminPort := zoneIngress.Spec.GetNetworking().GetAdmin().GetPort(); adminPort != 0 {
+			params.AdminPort = adminPort
+		} else {
+			if request.AdminPort != 0 {
+				// backwards compatibility, Inspect API may not work properly
```

Comment:
"Inspect API may not work properly" 

in what way? 🤔

Reply:
That's about ability to get `/config_dump` through Inspect API. If port is set through `--admin-port` flag then it will work for compatibility reasons, but new API wont work because we don't update DPP resource on the fly with `networking.admin.port`

Reply:
ok, could you clarify the comment in the code?